feat/feature.go

```diff
@@ -105,6 +105,24 @@ type Collection interface {
 	Location() Feature
 }
 
+// BasePositionOf returns the position in f converted to coordinates relative
+// to the shallowest nil feature location and a reference feature which will
+// be the deepest non nil feature. The reference feature can be used by
+// different calls to BasePositionOf to verify that returned positions are
+// comparable.
+// BasePositionOf will panic if the feature chain is deeper than 1000 links.
+//
+// or
+//
+// BasePositionOf returns the position in f converted to coordinates relative
+// to the shallowest nil feature location and a reference feature which will
+// be the parent feature of the swallowest nil feature. The reference feature
+// can be used by different calls to BasePositionOf to verify that returned
+// positions are comparable.
+// BasePositionOf will panic if the feature chain is deeper than 1000 links.
+//
+// or
+//
 // BasePositionOf returns the position in f converted to coordinates in the
```

Comment:
"... first non-nil feature location..." is not right. What about:
```
// BasePositionOf returns the position in f coordinates converted to coordinates
// relative to the first nil feature location, and a reference which is the feature location
// preceding the nil. The returned reference feature should be used by
// callers of BasePositionOf to verify that coordinates are comparable.
// BasePositionOf will panic if the feature chain is deeper than 1000 links.
```
Note that I avoid the word "reference" when describing the first part of the return because this creates the mental connection that the following is always true, which is not:
```go
basePos, ref := BasePositionOf(f, pos)
posWithin, _ := PositionWithin(f, ref, pos)
basePos == posWithin // this can be false - see the last one in baseCoordsTests
```
The returned feature is called a reference only because it can be used to ensure that coords are comparable.